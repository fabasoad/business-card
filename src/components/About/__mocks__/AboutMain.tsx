import * as React from 'react'

export default function AboutMain() {
  return (
    <div data-testid="AboutMain"></div>
  )
}
