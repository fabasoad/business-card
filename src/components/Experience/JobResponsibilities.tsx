import * as React from 'react'
import { useTranslation } from 'react-i18next'

interface JobResponsibilitiesProps {
  count: number
  i18nKeyPrefix: string
}

const JobResponsibilities: React.FunctionComponent<JobResponsibilitiesProps> =
  ({ count, i18nKeyPrefix }) => {
    const { t } = useTranslation()
    return (
      <>
        <h5>{t('business-card-experience-responsibilities-title')}</h5>
        <ul>
          {Array(count).fill(i18nKeyPrefix)
            .map((p, i) => `${p}-${i + 1}`)
            .map((key) => (<li key={key}>{t(key)}</li>))}
        </ul>
      </>
    )
  }

export default JobResponsibilities
