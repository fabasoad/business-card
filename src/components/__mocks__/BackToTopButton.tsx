import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
