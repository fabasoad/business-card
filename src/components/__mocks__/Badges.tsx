import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
