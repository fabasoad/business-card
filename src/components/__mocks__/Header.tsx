import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
