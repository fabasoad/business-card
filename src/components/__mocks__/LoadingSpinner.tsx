import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
