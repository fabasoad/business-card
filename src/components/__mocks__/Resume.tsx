import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
