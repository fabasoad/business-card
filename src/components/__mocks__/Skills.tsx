import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
