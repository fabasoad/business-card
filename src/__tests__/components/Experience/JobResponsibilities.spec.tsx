/// <reference types="jest" />
import * as React from 'react'
import { shallow } from 'enzyme'
import JobResponsibilities from '../../../components/Experience/JobResponsibilities'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeEach(() => {
  tMock = useTranslation().t
})

test('should render JobResponsibilities correctly', () => {
  const prefix = 'some-prefix'
  const count = 2
  const wrapper = shallow(<JobResponsibilities
    count={count}
    i18nKeyPrefix={prefix}
  />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toBeCalledWith('business-card-experience-responsibilities-title')
  for (let i = 1; i < count; i++) {
    expect(tMock).toHaveBeenNthCalledWith(i + 1, `${prefix}-${i}`)
  }
})
