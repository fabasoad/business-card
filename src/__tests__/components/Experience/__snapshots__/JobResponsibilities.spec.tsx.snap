// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render JobResponsibilities correctly 1`] = `
<Fragment>
  <h5>
    business-card-experience-responsibilities-title
  </h5>
  <ul>
    <li
      key="some-prefix-1"
    >
      some-prefix-1
    </li>
    <li
      key="some-prefix-2"
    >
      some-prefix-2
    </li>
  </ul>
</Fragment>
`;
